### PR TITLE
Approve and Deny companies

### PR DESCRIPTION
**Pull request by jefftrojan**

### CODE REVIEW

**Review by Irirwanirira — APPROVED** [review]

File: src/controllers/auth.js [hunk 1]
```diff
@@ -43,6 +59,7 @@ export const register = async (req, res) => {
       companyId: newUser.companyId,
       createdAt: newUser.createdAt,
       updatedAt: newUser.updatedAt,
+      // roleId: newUser.roleId,
```
Comment (alainburindi) [inline src/controllers/auth.js]: Would you mind removing commented codes?

File: src/controllers/auth.js [hunk 2]
```diff
@@ -67,7 +84,64 @@ export const register = async (req, res) => {
...
-      error.message
+      error.message,
     );
   }
 };
+
+export const ApproveCompanyActivateUser = async (req, res) => {
+  const { id } = req.params;
+  const { status } = req.body;
+  const user = await findUserById(id);
```
Comment (alainburindi) [inline src/controllers/auth.js]: Why do we need to specify the user ID in order to activate a company?

I also think this should be in the companies controller

File: src/controllers/auth.js [hunk 3]
```diff
@@ -67,7 +84,64 @@ export const register = async (req, res) => {
...
+        "Unable to update user and company",
+      );
+  } else {
+    // Handle other status values here, if necessary.
+    return formatResponse(
+      res,
+      StatusCodes.BAD_REQUEST,
+      null,
+      "Invalid status value",
+    );
```
Comment (alainburindi) [inline src/controllers/auth.js]: This can be handled by joi validator

File: src/controllers/auth.js [hunk 4]
```diff
@@ -67,7 +84,64 @@ export const register = async (req, res) => {
...
+    const updatedCompany = await updateCompany(company.id, {
+      status: "inactive",
+    });
+    if (!updatedUser || !updatedCompany)
+      return formatResponse(
+        res,
+        StatusCodes.BAD_REQUEST,
+        null,
+        "Unable to update user and company",
+      );
```
Comment (alainburindi) [inline src/controllers/auth.js]: We have too many duplicates here. Can we make it better? What do you think?

**Review by alainburindi — CHANGES_REQUESTED** [review]

File: src/controllers/auth.js (hunk 1 above)
Comment (jefftrojan) [reply src/controllers/auth.js]: Done

File: src/controllers/auth.js (hunk 4 above)
Comment (jefftrojan) [reply src/controllers/auth.js]: resolved

File: src/controllers/auth.js (hunk 3 above)
Comment (jefftrojan) [reply src/controllers/auth.js]: Resolved as well

**Review by alainburindi — CHANGES_REQUESTED** [review]
I can't see any logic to approve or deny company

**Review by mwafrika — APPROVED** [review]

**Pull request closed without merging**